hoot-core/src/main/cpp/hoot/core/auth/HootServicesLoginManager.cpp

```diff
@@ -0,0 +1,220 @@
+    "application." << std::endl;
+  std::cout << "2. Grant Hootenanny access to the application." << std::endl;
+  std::cout << "3. Copy and paste your verifier code from the browser here and press ENTER: ";
+  std::getline(std::cin, consoleInput);
+  const QString verifier = QString::fromStdString(consoleInput);
+  LOG_VARD(verifier);
+  return verifier;
+}
+
+long HootServicesLoginManager::verifyUserAndLogin(const QString requestToken,
+                                                  const QString verifier, QString& userName)
+{
+  QUrl loginUrl;
+  HootNetworkRequest loginRequest = _getLoginRequest(requestToken, verifier, loginUrl);
+  try
+  {
+    loginRequest.networkRequest(loginUrl.toString());
+  }
+  catch (const std::exception& e)
+  {
+    const QString exceptionMsg = e.what();
+    throw HootException("Error verifying user. error: " + exceptionMsg);
+  }
+  if (loginRequest.getHttpStatus() != 200)
+  {
+    throw HootException("Error verifying user. error: " + loginRequest.getErrorString());
+  }
+  LOG_VART(_cookies->size());
+  LOG_VART(_cookies->toString());
+
+  // reply contains a user object
+  const long userId = _parseLoginResponse(loginRequest.getResponseContent());
+
+  HootApiDb db;
+  //hoot db requires a layer to open, but we don't need one here...so put anything in
+  QUrl url(HootApiDb::getBaseUrl().toString() + "/blah");
+  db.open(url);
+  userName = db.getUserNameById(userId);
+
+  return userId;
+}
+
+HootNetworkRequest HootServicesLoginManager::_getLoginRequest(const QString requestToken,
+                                                              const QString verifier,
+                                                              QUrl& loginUrl) const
+{
+  HootNetworkRequest loginRequest;
+  LOG_VART(_cookies->size());
+  LOG_VART(_cookies->toString());
+  loginRequest.setCookies(_cookies);
+  LOG_VART(ConfigOptions().getHootServicesAuthVerifyEndpoint());
+
+  loginUrl.setUrl(ConfigOptions().getHootServicesAuthVerifyEndpoint());
+  loginUrl.addQueryItem("oauth_token", QString(QUrl::toPercentEncoding(requestToken)));
+  loginUrl.addQueryItem("oauth_verifier", QString(QUrl::toPercentEncoding(verifier)));
+  LOG_VART(loginUrl.toString());
+
+  return loginRequest;
+}
+
+long HootServicesLoginManager::_parseLoginResponse(const QString response) const
+{
+  LOG_VART(response);
+  boost::shared_ptr<boost::property_tree::ptree> replyObj =
+    StringUtils::jsonStringToPropTree(response);
+  const long userId = replyObj->get<long>("id");
+  LOG_VARD(userId);
+  return userId;
+}
+
+void HootServicesLoginManager::getAccessTokens(const long userId, QString& accessToken,
+                                               QString& accessTokenSecret)
+{
+  HootApiDb db;
+  LOG_VARD(HootApiDb::getBaseUrl());
+  //hoot db requires a layer to open, but we don't need one here...so put anything in
+  QUrl url(HootApiDb::getBaseUrl().toString() + "/blah");
```

Comment:
Haha! Interesting choice of layer name!

Reply:
Yeah, its a little silly.  HootApiDb probably needs some refactoring of sorts regarding that.